### PR TITLE
Be pay route #3

### PR DESCRIPTION
**Pull request by lipob**
### What this PR does (required):
- Payment route created.
- For this issue I needed to update Profile model and add "customerId", "serviceId" and "hourlyRate" properties.
- If the user hasn't a customer ID, tha app creates a new stripe customer and updates the user profile.
- If the pet sitter hasn't a registered service ID, it creates a new service related to the sitter and updates its profile.
- Then creates a price and a new checkout session.

### CODE REVIEW

**Review by ajaytomgeorge — COMMENTED** [review]
@lipob Minor comments from my side, though I would have loved to see some screenshots which limited frontend review, @moffatethan might be able to give some more comments on this

File: server/controllers/request.js
Comment (lipob) [reply server/controllers/request.js:87]: I added it to the arguments just to be consistent with the rest of the controllers

File: server/controllers/request.js
Comment (lipob) [reply server/controllers/request.js:134]: Ok. I'll pass the URLs within req.body. Thanks

**Comment by lipob** [comment]
Fixed all conflicts and code review. Waiting for approval. Thanks.

File: server/controllers/request.js
Comment (moffatethan) [reply server/controllers/request.js:105]: If we aren't using it then it can be removed, arguments aren't really a huge deal in terms of consistency with the overall project. Things like using const vs function is consistency, using async/await instead of .then, etc.

File: server/controllers/request.js
Comment (moffatethan) [inline server/controllers/request.js:135]: This is a potential flaw in our logic, if someone malicious intentionally hits without serviceId then they can bloat our Stripe products with unnecessary products.

A better option is just making sure there is a product available that we've created and then returning an error if it cannot be retrieved.

File: server/controllers/request.js
Comment (moffatethan) [inline server/controllers/request.js:126]: Same as below for here as well

File: server/controllers/request.js
Comment (moffatethan) [inline server/controllers/request.js:156]: Nice use of Stripe Checkout 👍🏻

**Review by moffatethan — COMMENTED** [review]
Good work Leopoldo, left you some comments to address. Thank you!

File: server/controllers/request.js
Comment (lipob) [reply server/controllers/request.js:135]: Hi Ethan. Should I create a new Stripe product when user becomes a sitter? So, doing it that way then when a customer wants to pay, the sitter information stored in the DB provides the product ID.  Please let me know if its right and I do this change. Thanks.

File: server/controllers/request.js
Comment (lipob) [reply server/controllers/request.js:126]: I could create a new Stripe customer when a user registers. That's right?

File: server/controllers/request.js
Comment (lipob) [reply server/controllers/request.js:156]: Thanks

File: client/src/components/BookingCards/BookingCard.tsx
Comment (lipob) [reply client/src/components/BookingCards/BookingCard.tsx]: Thank you. I will try to fix it in thw bookings branch

File: client/src/App.tsx
Comment (moffatethan) [inline client/src/App.tsx:48]: Let's fix this please, these are merge conflicts and they were not changed properly and would crash our code.

File: server/controllers/payment.js
Comment (moffatethan) [inline server/controllers/payment.js:27]: A proper error description here please.

```suggestion
        throw new Error('User already has a customer Id');
```

**Review by moffatethan — COMMENTED** [review]
Thanks @lipob, left you some comments on this PR.

File: client/src/App.tsx
Comment (lipob) [reply client/src/App.tsx:48]: Oh, I've fixed it. Thanks

**Comment by lipob** [comment]
Hello @moffatethan When I was creating the Payment component in the FE I realized that it's related to add a new payment method, so before merge, I'll have to create a new controller in this branch to create the payment method and attach it to the user

**Review by moffatethan — APPROVED** [review]

**Pull request merged**
